### DISCUSSION TITLE
Moordyn：About output parameters.

### DISCUSSION THREAD

**Discussion by LeoHongLiZuo** [OpenFAST/openfast#2175] [Q&A]
Dear guys,
I want to analyze the stiffness curves (surge, sway, and yaw stiffness as functions of offset/rotation angles) of the mooring line after attaching clump weight on the mooring line.
But I don't know how to output the resultant force and the vertical moment of the mooring system in the MoorDyn.
I can't find the description in the MoorDyn Users Guide
Does anyone know how to export the resultant force and the vertical moment of the mooring system?
Or does anyone know how to get these stiffness curves from OpenFAST output？
Your sincerely，
Leo

**Comment by jjonkman** [comment]
Dear @LeoHongLiZuo,

You can output the X/Y/Z components of the force and moment at each fairlead in MoorDyn via the outputs available through the `OutList`--see the MoorDyn v2 documentation available here: https://moordyn.readthedocs.io/en/latest/inputs.html#id5.  You would then have to sum these forces and moments to arrive at the total reaction forces and moments of the mooring system.

I'm not sure how easy it is to calculate quasi-static force-displacement curves or a linearized stiffness matrix of the mooring system in MoorDyn, which naturally considers dynamics (perhaps MoorDyn experts can chime in), but if the quasi-static mooring response is your goal, quasi-static models such as MAP++ or MoorPy are likely better mooring modules to use.

Best regards,

**Reply by LeoHongLiZuo** [reply]
Dear Jonkman,
Thank you for your patient reply!
I would like to ask whether the specified directions of the X components of the force at different fairlead points in the Outlist are the same?
If not, what are the regulations?
In addition, I would like to ask where the moment calculation formula in Outlist can be consulted.
Yours sincerely,
Leo

**Reply by RyanDavies19** [reply]
Hi @LeoHongLiZuo, I can chime in and hopefully answer your questions. Yes all the X directions in outputs are the same direction. All outputs in MoorDyn are output in the global frame, the same reference frame you define your mooring system locations at in the input. 

If you are looking for the specific moment calculation on a 6dof of freedom object (body or rod) that gets output, it gets a bit tricky. You can see in the [MoorDyn object hierarchy](https://moordyn.readthedocs.io/en/latest/structure.html#model-structure) that bodies and rods receive the forces from objects attached to them at a connection point. These forces are then translated to a 6DOF force about the bodies center (as initially defined in the MD input file). A simple example of this process can be seen in MoorDyn_Body.f90 in the [Body_DoRHS subroutine](https://github.com/OpenFAST/openfast/blob/6a7a543790f3cad4a65b87242a619ac5b34b4c0f/modules/moordyn/src/MoorDyn_Body.f90#L374). Once all of the attached objects forces on the 6DOF object are computed, those moments are output. The 6DOF force vector is then integrated to solve for the kinematics of the system.

**Reply by LeoHongLiZuo** [reply]
Dear @RyanDavies19,
Thank you very much for your answer！
I tried some experiments like you said.
There's one more question I'd like to ask.
The figure above is from“Definition of the Semisubmersible Floating System for Phase II of OC4” in 2014.
![图片](https://github.com/OpenFAST/openfast/assets/108517989/e8af7b2c-6f07-4a21-9117-f669d3ef1956)
I think Moordyn V1 was used at that time.
May I ask how the Restoring Moment changed with Yaw in the figure is derived？
Is it through calculation or Openfast output? I want to reproduce this graph
Look forward to hearing from you！
Yours sincerely,
Leo

**Reply by RyanDavies19** [reply]
@LeoHongLiZuo I will defer to @jjonkman on this one, because to my knowledge MoorDyn was not integrated into OpenFAST until 2015 (correct me if I'm wrong @jjonkman).

**Reply by jjonkman** [reply]
Actually, the nonlinear mooring system force-displacement relationships for the OC4-DeepCwind semisubmersible were calculated in the old quasi-static mooring model of FAST that was replaced with MAP++ in FAST v8 and OpenFAST.  You should be able to calculate the equivalent now with MAP++.

Best regards,

**Reply by LeoHongLiZuo** [reply]
Dear @RyanDavies19 
Thanks again for your help.
I have one more question about Moordyn in this process.
![图片](https://github.com/OpenFAST/openfast/assets/108517989/dce16425-edc8-409a-b434-ec42bd88385c)
I wonder if this M(node mass) refers to mass in water or mass in air？
Your sincerely，
Leo

**Reply by RyanDavies19** [reply]
@LeoHongLiZuo that is the mass in air
